### PR TITLE
Add new job : check local swift dependencies

### PR DESCRIPTION
**Pull request by GErP83**

### CODE REVIEW

**Comment by FranzBusch** [comment]
Could you explain what the intention behind this check is?

**Comment by GErP83** [comment]
Sometimes in Swift development, we use local dependencies, such as:
```
dependencies: [
        .package(path: "../SomePackage")
    ],
```
However, we might forget to remove or replace them later. This script can help by providing a warning in such cases.

File: .github/workflows/soundness.yml [hunk 1]
```diff
@@ -51,6 +51,10 @@ on:
         type: string
         description: "Container image for the shell check job. Defaults to latest Swift Ubuntu image."
         default: "swift:6.0-noble"
+      local_swift_dependencies_check_enabled:
+        type: boolean
+        description: "Boolean to enable the local swift dependencies check job. Defaults to true."
+        default: true
```
Comment (ahoppen) [inline .github/workflows/soundness.yml:57]: All the Swift packages that are built into the toolchain will have local Swift dependencies, eg. https://github.com/swiftlang/swift-format/blob/d143a8a59c89aba5bdadf96787f7f41197ce6b71/Package.swift#L173-L175. I’m thus not sure if we want to enable this by default.

Comment (tib) [reply .github/workflows/soundness.yml:57]: I agree, this should be disabled by default. Users should be able to opt-in for this check. Apart from that I think this is very useful in some cases. :)

Comment (GErP83) [reply .github/workflows/soundness.yml]: Changed the the default value to **false**.

**Review by FranzBusch — CHANGES_REQUESTED** [review]
As @ahoppen pointed out there are use-cases where local dependencies are expected such as packages used in the toolchain. Additionally, a lot of repositories are using nested packages that reference the root package for benchmarks, examples and much more. 

At this point, I don't see a real need to enforce no local package dependencies across our repositories and would like us to wait with this until we see it becoming a problem.

**Pull request closed without merging**